projects/plugins/protect/src/class-status.php

```diff
@@ -7,52 +7,39 @@
 
 namespace Automattic\Jetpack\Protect;
 
-use Automattic\Jetpack\Connection\Client;
-use Automattic\Jetpack\Connection\Manager as Connection_Manager;
-use Automattic\Jetpack\Plugins_Installer;
-use Automattic\Jetpack\Sync\Functions as Sync_Functions;
-use Jetpack_Options;
-use WP_Error;
+use Jetpack_Plan;
 
 /**
  * Class that handles fetching and caching the Status of vulnerabilities check from the WPCOM servers
  */
 class Status {
```

Comment:
Did you consider extracting the common methods from `Status` to something like `Status_General`, and making `Status` abstract?
I think that would improve the code structure and make it easier to maintain.

Reply:
Mentioned in my comment above, I'm new to abstract classes so I may be missing a few things here!

I'm not too sure which methods I would want to move to or keep in the abstract `Status` class:
* Shared methods such as `get_plugins_threats()` or `should_use_cache()` could stay in the `abstract Status` class, as non-abstract standard methods. Is that correct? Or would that be bad practice?
* I'm not sure if the methods currently defined in `Protect_Status` and `Scan_Status` would be good candidates for a static method definition, despite the repetition of method names. Methods such as `get_api_url()`, `fetch_from_api()`, and `normalize_api_data()` are repeated across these two child classes, but a third `Status` might not follow the same "fetch from API and normalize data" flow that the current Status child classes follow, if that makes sense. Due to that, it doesn't feel right to define these methods in `Status` to me, which is why I define `get_status()` in the parent class, but don't go any deeper.

Overall, you probably have a good point that should be pursued, I am just not quite fully understanding the benefit yet due to my lack of static experience 😅 Any additional insight would really be appreciated! Thanks again @sergeymitr :+1: